src/resources/parser/glb-parser.js

```diff
@@ -1367,29 +1367,107 @@ const createAnimation = function (gltfAnimation, animationIndex, gltfAccessors,
         const node = nodes[target.node];
         const entityPath = constructNodePath(node);
-        curve._paths.push({
+
+        if (target.path.startsWith('weights')) {
+            // create a curve for each morph target weight present in the animation data
+            const morphTargetCount = outputMap[curve.output].data.length / inputMap[curve.input].data.length;
+            const keyframeCount = outputMap[curve.output].data.length / morphTargetCount;
+
+            for (let j = 0; j < morphTargetCount; j++) {
+                const morphTargetOutput = new Float32Array(keyframeCount);
+                // the output data for all morph targets in a single curve is interleaved. We need to retrieve the keyframe output data for a single morph target
+                for (let k = 0; k < keyframeCount; k++) {
+                    morphTargetOutput[k] = outputMap[curve.output].data[k * morphTargetCount + j];
+                }
+                const output = new AnimData(1, morphTargetOutput);
+                // add the individual morph target output data to the outputMap using a negative value key (so as not to clash with sampler.output values)
+                outputMap[-outputCounter] = output;
+                const morphCurve = {
+                    paths: [{
+                        entityPath: entityPath,
+                        component: 'graph',
+                        propertyPath: [`weight.${retrieveWeightName(node.name, j)}`]
+                    }],
+                    // each morph target curve input can use the same sampler.input from the channel they were all in
+                    input: curve.input,
+                    // but each morph target curve should reference its individual output that was just created
+                    output: -outputCounter,
+                    interpolation: curve.interpolation
+                };
+                outputCounter++;
+                // add the morph target curve to the curveMap
+                curveMap[`morphCurve-${i}-${j}`] = morphCurve;
+            }
+            // after all morph targets in this curve have been included in the curveMap, this curve and its output data can be deleted
+            delete curveMap[channel.sampler];
+            delete outputMap[curve.output];
+            continue;
+        }
+
+        curve.paths.push({
             entityPath: entityPath,
             component: 'graph',
             propertyPath: [transformSchema[target.path]]
         });
+    }
+
+    const inputs = [];
+    const outputs = [];
+    const curves = [];
+    let keys;
+
+    // Add each input in the map to the final inputs array. The inputMap should now reference the index of input in the inputs array instead of the input itself.
+    keys = Object.keys(inputMap);
+    for (i = 0; i < keys.length; i++) {
+        const inputKey = keys[i];
+        const input = inputMap[inputKey];
+        inputs.push(input);
+        inputMap[inputKey] = inputs.length - 1;
+    }
```

Comment:
Actually, instead of using `Object.keys`, why not just do something like:

```
    for (const inputKey in inputMap) {
        inputs.push(inputMap[inputKey]);
        inputMap[inputKey] = inputs.length - 1;
    }
```